#746 training dummy

PhantomRyser opened the pull request:
Closes #746

**Notes for the reviewer:**

Dummy Titan has been raised to what I hope is the new standard with added functionality. Will now detect the player and turn towards him. If the player leaves the view distance declared the titan will stop rotating. The titan can be disabled via the ankle and killed via the nape. Will reset after ~6 seconds to be targeted again.

**Contributor guidelines:**
1. A build has been uploaded to the discord #builds channel & has been tested by the testing team
2. Assure the code is inline with the [AoTTG2 coding conventions](https://github.com/AoTTG-2/AoTTG-2/wiki/Code-&-Style-Conventions)
3. Fix any issues that the CI reports (Sonarcloud & unit tests)

After all above conditions are met, @Lithrun will review your Pull Request. Resolve any comments of the reviewer or discuss them if you disagree.

PhantomRyser committed:
Added Multiplayer Functionality to the Dummy Titan

Should update for all Photon members including host. Just need to make the ankle and Hero detection work outside singleplayer

PhantomRyser committed:
Forgot some changes that needed to be made

PhantomRyser committed:
Got rid of new gamemode I added

Was breaking the game lol

PhantomRyser committed:
Removed Some Unnecessary Code

PhantomRyser committed:
Multiplayer Update for Rotating the Titan

Still needs some work. Dummy Rotates when it shouldn't. My guess is I'm missing a PunRPC call somewhere

PhantomRyser committed:
Finalize the Dummy

PhantomRyser committed:
Multiplayer Fix

PhantomRyser committed:
Added comments to code and change spawn behavior

Now can see the titan when placed in scene to make it easier to develop levels. The place holder will be destroyed on load and remade for networking purposes.

PhantomRyser committed:
Removed DummySpawn from Forest Map

PhantomRyser committed:
Sonar Cloud Fixes

Added namespaces for the scripts and removed some unnecessary code.

PhantomRyser committed:
Fix for SonarCloud

PhantomRyser committed:
Reversed the changes to the  game mode. Figured I didn't actually need to touch this

Lithrun commented:
On Hold as per requested by @dark3art, remove the tag and give me a ping once it can be reviewed.

dark3art committed:
added sounds on the Training dummy's

PhantomRyser committed:
Merge branch 'development' into #746-Training-Dummy

PhantomRyser committed:
Development Merge

Merged Development into Training Dummy Branch

dark3art committed:
Merge branch 'development' into #746-Training-Dummy

PhantomRyser committed:
Merge branch '#746-Training-Dummy' of https://github.com/AoTTG-2/AoTTG-2 into #746-Training-Dummy

PhantomRyser committed:
Update on Dummy Titan Sound

Added the wiggle sound as Dark wanted for when the Dummy Titan is "chasing" the player. Also made sure it worked on multiplayer but can be tested to verify the solution works

dark3art committed:
tweaked some sounds a bit and fixed compiler error

dark3art committed:
Merge branch 'development' into #746-Training-Dummy

dark3art committed:
fixed some issue's on the dummy titan

dark3art committed:
Merge branch 'development' into #746-Training-Dummy

dark3art committed:
fixed dummy fall sound

dark3art committed:
Merge branch 'development' into #746-Training-Dummy

dark3art committed:
fixed SFX sounds

dark3art committed:
fixed issue with basic attack

dark3art committed:
code smells fixes

dark3art committed:
Merge branch 'development' into #746-Training-Dummy

PhantomRyser committed:
Merge branch 'development' into #746-Training-Dummy

PhantomRyser committed:
Adjusted the Animations

Now the falling and rising animations more closely matches the sounds for those specific actions. I'm not animator so it ain't perfect but hardly noticeable

PhantomRyser committed:
Slight Adjustments
